File: vul.java
import java.io.*;
import java.sql.*;
import java.util.Random;

public class VulnerableExample {

    // Hardcoded credentials
    private static final String DB_USER = "admin";
    private static final String DB_PASS = "password123";

    public static void main(String[] args) {
        String userInput = args.length > 0 ? args[0] : "default";

        sqlInjectionExample(userInput);
        commandInjectionExample(userInput);
        insecureRandomExample();
        pathTraversalExample(userInput);
    }

    public static void sqlInjectionExample(String userInput) {
        try {
            Connection conn = DriverManager.getConnection("jdbc:mysql://localhost:3306/test", DB_USER, DB_PASS);
            Statement stmt = conn.createStatement();

            // SQL Injection vulnerability
            String query = "SELECT * FROM users WHERE username = '" + userInput + "'";
            ResultSet rs = stmt.executeQuery(query);

            while (rs.next()) {
                System.out.println("User found: " + rs.getString("username"));
            }

            conn.close();
        } catch (SQLException e) {
            e.printStackTrace();
        }
    }

    public static void commandInjectionExample(String input) {
        try {
            // Command Injection vulnerability
            String command = "ping -c 1 " + input;
            Process proc = Runtime.getRuntime().exec(command);

            BufferedReader reader = new BufferedReader(new InputStreamReader(proc.getInputStream()));
            String line;
            while ((line = reader.readLine()) != null) {
                System.out.println(line);
            }

        } catch (IOException e) {
            e.printStackTrace();
        }
    }

    public static void insecureRandomExample() {
        // Insecure randomness
        Random rand = new Random();
        int token = rand.nextInt();
        System.out.println("Generated token: " + token);
    }

    public static void pathTraversalExample(String filename) {
        try {
            // Path traversal vulnerability
            File file = new File("/var/data/" + filename);
            BufferedReader reader = new BufferedReader(new FileReader(file));

            String line;
            while ((line = reader.readLine()) != null) {
                System.out.println("Line: " + line);
            }
            reader.close();

        } catch (IOException e) {
            e.printStackTrace();
        }
    }
}

File: vul.cs
using System;
using System.Data.SqlClient;
using System.Diagnostics;
using System.IO;

namespace VulnerableApp
{
    class Program
    {
        // Hardcoded credentials – CWE-798
        private static string dbUser = "admin";
        private static string dbPass = "pass123";

        static void Main(string[] args)
        {
            string userInput = args.Length > 0 ? args[0] : "default";

            SqlInjection(userInput);
            CommandInjection(userInput);
            PathTraversal(userInput);
            InsecureLogging(userInput);
        }

        // SQL Injection – CWE-89
        static void SqlInjection(string input)
        {
            string connString = $"Server=localhost;Database=TestDB;User Id={dbUser};Password={dbPass};";
            SqlConnection conn = new SqlConnection(connString);
            conn.Open();

            string query = $"SELECT * FROM Users WHERE Username = '{input}'";
            SqlCommand cmd = new SqlCommand(query, conn);
            SqlDataReader reader = cmd.ExecuteReader();

            while (reader.Read())
            {
                Console.WriteLine($"User: {reader["Username"]}");
            }

            conn.Close();
        }

        // Command Injection – CWE-78
        static void CommandInjection(string input)
        {
            string command = "ping " + input;
            Process.Start("cmd.exe", "/C " + command);  // Unsafe input execution
        }

        // Path Traversal – CWE-22
        static void PathTraversal(string filename)
        {
            string filePath = @"C:\Users\Public\" + filename;
            if (File.Exists(filePath))
            {
                string content = File.ReadAllText(filePath);
                Console.WriteLine("File content: " + content);
            }
            else
            {
                Console.WriteLine("File not found");
            }
        }

        // Insecure Logging of sensitive input – CWE-532
        static void InsecureLogging(string input)
        {
            File.AppendAllText("log.txt", "User input: " + input + Environment.NewLine);
        }
    }
}
